### ISSUE TITLE
Query: NullReferenceException when using multiple navigation properties and comparing to null

### ISSUE THREAD

**Issue by Wain123** [dotnet/efcore#12062]
I have entities LgrBewegung, LgrLagercharge and LgrKonfektion.
LgrBewegung has a navigation property to LgrLagercharge, configured like this:
```C#
builder.HasOne(t => t.Lagercharge)
.WithMany("CLgrBewegungLagercharge").HasForeignKey("Lagercharge_Seqnr")
.OnDelete(DeleteBehavior.Restrict);
```

LgrLagercharge has a navigation property to LgrKonfektion, configured the same way:
```C#
builder.HasOne(t => t.Konfektion)
.WithMany("CLgrLagerchargeKonfektion").HasForeignKey("Konfektion_Seqnr")
.OnDelete(DeleteBehavior.Restrict);
```

The following query throws a NullReferenceException:
```C#
LgrKonfektion k = null;
ctx.LgrBewegung.Where(b => b.Lagercharge.Konfektion == k).Count();
```

EF Core version: 2.1.0-rc1-30677
Database Provider: Microsoft.EntityFrameworkCore.Oracle
Operating system: Windows 10
IDE: Visual Studio Professional 2017 15.6.5

Stack trace:
```
 	[Lightweight Function]	
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Metadata.Internal.ClrPropertyGetter<PI.PI_STD.DataAccess.BO.Lgr.LgrKonfektion, decimal?>.GetClrValue(object instance = null)	Unknown
 	Microsoft.EntityFrameworkCore.Relational.dll!Microsoft.EntityFrameworkCore.Storage.Internal.TypeMappedPropertyRelationalParameter.AddDbParameter(System.Data.Common.DbCommand command = {Oracle.DataAccess.Client.OracleCommand}, object value = null)	Unknown
 	Microsoft.EntityFrameworkCore.Relational.dll!Microsoft.EntityFrameworkCore.Storage.Internal.RelationalParameterBase.AddDbParameter(System.Data.Common.DbCommand command = {Oracle.DataAccess.Client.OracleCommand}, System.Collections.Generic.IReadOnlyDictionary<string, object> parameterValues = Count = 1)	Unknown
>	Parallel.EFCore.Oracle.dll!Microsoft.EntityFrameworkCore.Oracle.Storage.Internal.OracleRelationalCommandBuilderFactory.OracleRelationalCommandBuilder.OracleRelationalCommand.CreateCommand(Microsoft.EntityFrameworkCore.Storage.IRelationalConnection connection = {Microsoft.EntityFrameworkCore.Oracle.Storage.Internal.OracleRelationalConnection}, System.Collections.Generic.IReadOnlyDictionary<string, object> parameterValues = Count = 1) Line 305	C#
 	Parallel.EFCore.Oracle.dll!Microsoft.EntityFrameworkCore.Oracle.Storage.Internal.OracleRelationalCommandBuilderFactory.OracleRelationalCommandBuilder.OracleRelationalCommand.Execute(Microsoft.EntityFrameworkCore.Storage.IRelationalConnection connection = {Microsoft.EntityFrameworkCore.Oracle.Storage.Internal.OracleRelationalConnection}, Microsoft.EntityFrameworkCore.Diagnostics.DbCommandMethod executeMethod = ExecuteReader, System.Collections.Generic.IReadOnlyDictionary<string, object> parameterValues = Count = 1) Line 81	C#
 	Microsoft.EntityFrameworkCore.Relational.dll!Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(Microsoft.EntityFrameworkCore.Storage.IRelationalConnection connection = {Microsoft.EntityFrameworkCore.Oracle.Storage.Internal.OracleRelationalConnection}, System.Collections.Generic.IReadOnlyDictionary<string, object> parameterValues = Count = 1)	Unknown
 	Microsoft.EntityFrameworkCore.Relational.dll!Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable<Microsoft.EntityFrameworkCore.Storage.ValueBuffer>.Enumerator.BufferlessMoveNext(Microsoft.EntityFrameworkCore.DbContext _ = {PI.PI_ADM.DataAccess.Generated.Context.Context}, bool buffer = false)	Unknown
 	Parallel.EFCore.Oracle.dll!Microsoft.EntityFrameworkCore.Oracle.Storage.Internal.OracleExecutionStrategy.Execute<bool, bool>(bool state = false, System.Func<Microsoft.EntityFrameworkCore.DbContext, bool, bool> operation = {Method = {System.Reflection.RuntimeMethodInfo}}, System.Func<Microsoft.EntityFrameworkCore.DbContext, bool, Microsoft.EntityFrameworkCore.Storage.ExecutionResult<bool>> verifySucceeded = null) Line 31	C#
 	Microsoft.EntityFrameworkCore.Relational.dll!Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable<Microsoft.EntityFrameworkCore.Storage.ValueBuffer>.Enumerator.MoveNext()	Unknown
 	Microsoft.EntityFrameworkCore.Relational.dll!Microsoft.EntityFrameworkCore.Query.QueryMethodProvider.GetResult<int>(System.Collections.Generic.IEnumerable<Microsoft.EntityFrameworkCore.Storage.ValueBuffer> valueBuffers = {Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable<Microsoft.EntityFrameworkCore.Storage.ValueBuffer>}, bool throwOnNullResult = false)	Unknown
 	[Lightweight Function]	
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ResultEnumerable<int>.GetEnumerator()	Unknown
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor<int>.EnumeratorExceptionInterceptor.MoveNext()	Unknown
 	System.Core.dll!System.Linq.Enumerable.First<int>(System.Collections.Generic.IEnumerable<int> source)	Unknown
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore.AnonymousMethod__0(Microsoft.EntityFrameworkCore.Query.QueryContext qc = {Microsoft.EntityFrameworkCore.Query.RelationalQueryContext})	Unknown
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute<int>(System.Linq.Expressions.Expression query = {System.Linq.Expressions.MethodCallExpressionN})	Unknown
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.Execute<int>(System.Linq.Expressions.Expression expression = {System.Linq.Expressions.MethodCallExpressionN})	Unknown
 	System.Core.dll!System.Linq.Queryable.Count<PI.PI_STD.DataAccess.BO.Lgr.LgrBewegung>(System.Linq.IQueryable<PI.PI_STD.DataAccess.BO.Lgr.LgrBewegung> source)	Unknown
 	PI.PI_ADM.BuildEF.exe!EfTest.Program.Main(string[] args = {string[0]}) Line 28	C#
```

**Comment by tuespetre** [comment]
It's probably because it's trying to access the PK value of `k`. A workaround for now could be to explicitly compare the PK values of the objects:

```
LgrKonfektion k = null;
var pk = k?.PK;
ctx.LgrBewegung.Where(b => b.Lagercharge.Konfektion.PK == pk).Count();
```

The translation of entity equality comparison when one side is parameterized will probably have to be adjusted to expand out like `k != null && b.Lagercharge.Konfektion.PK == k.PK`.

**Comment by Wain123** [comment]
I just noticed that you don't need multiple navigation properties, the exception also happens if you compare a single navigation property to null. Sorry about the non-minimal repro.

**Comment by maumar** [comment]
currently blocked by SelectMany translation

**Comment by maumar** [comment]
duplicate of https://github.com/aspnet/EntityFrameworkCore/issues/15855
